src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,20 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int REQUIRED_AGE = 35;
+    private static final int REQUIRED_PERIODS_IN_UKR = 10;
+    private static final int LAST_YEAR_IN_UKR_INDEX = 1;
+    private static final int FIRST_YEAR_IN_UKR_INDEX = 0;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= REQUIRED_AGE && candidate.isAllowedToVote()
+                && candidate.getNationality().equals("Ukrainian")
+                && Integer.parseInt(candidate.getPeriodsInUkr().split("-")[LAST_YEAR_IN_UKR_INDEX])
+                - Integer.parseInt(candidate.getPeriodsInUkr().split("-")[FIRST_YEAR_IN_UKR_INDEX])
```

Comment:
```suggestion
        String[] splitedperiodInUkr = candidate.getPeriodsInUkr().split("-");
        return candidate.getAge() >= REQUIRED_AGE && candidate.isAllowedToVote()
                && candidate.getNationality().equals("Ukrainian")
                && Integer.parseInt(splitedperiodInUkr[LAST_YEAR_IN_UKR_INDEX])
                - Integer.parseInt(splitedperiodInUkr[FIRST_YEAR_IN_UKR_INDEX])
```

split just once and use those values